v2/yarpcjson/outbound.go

```diff
@@ -40,11 +41,11 @@ type Client struct {
 
 // Call performs an outbound JSON request.
 //
-// resBodyOut is a pointer to a value that can be filled with
-// json.Unmarshal.
+// resBodyOut and errorDetailsOut are pointers to a value that can be filled
+// with json.Unmarshal. errorDetailsOut are used for application errors.
 //
 // Returns the response or an error if the request failed.
-func (c Client) Call(ctx context.Context, procedure string, reqBody interface{}, resBodyOut interface{}, opts ...yarpc.CallOption) error {
+func (c Client) Call(ctx context.Context, procedure string, reqBody interface{}, resBodyOut interface{}, errorDetailsOut interface{}, opts ...yarpc.CallOption) error {
```

Comment:
Discussed offline - 

This is a little strange to see the `interface{}` used for both `resBodyOut` and `errorDetailsOut`, but this is necessary in JSON as a result of the new details API, i.e. users can set their details to whatever they like. If users manually configure their details to marshal into a specific type, they would pass that into the `Call` here.